### ISSUE TITLE
Prisma 2 integration / examples

### ISSUE THREAD

**Issue by lewebsimple** [Urigo/graphql-modules#1112]
Since Prisma 2 is now in its fourth beta release, it would be nice to follow up on #209 concerning prisma2 integration and maybe have some official examples.

There are many ways to approach this, because prisma is now a query builder and doesn't provide GraphQL schema building OOTB.

I see the following possibilities:
- Basic integration with @prisma/client provided in the context
- Code-first with [@nexus/schema](https://www.nexusjs.org/#/)
- Code-first with [typegraphql-prisma](https://www.npmjs.com/package/typegraphql-prisma)

Once Prisma2 SDK is ready, it should be possible to use a pure code-first approach, i.e. generate `schema.prisma` from type-graphql models (see https://github.com/MichalLytek/type-graphql/issues/476#issuecomment-618959888) which looks very promising!

**Comment by AhmedElywa** [comment]
@lewebsimple  I have a tool to auto generate types and crud system for graphql sdl first projects 
is this tool can help here. 
https://github.com/AhmedElywa/prisma-tools/tree/master/packages/sdl

I am not using graphql-modules and don't know how it's work but maybe it's help

if it's important and needed some changes to make it more helpful i can make it

**Comment by AhmedElywa** [comment]
Can you do example with one model how his file structure look like and i will generate tool to do this from schema.prisma

**Comment by AhmedElywa** [comment]
Maybe all i need to do to make my tool working perfect here is add 

Index.ts file inside every model folder because i already added `typeDefs.graphql` and `resolvers.ts`

```ts
import { GraphQLModule } from '@graphql-modules/core';
import * as typeDefs from './typeDefs.graphql';
import resolvers from './resolvers';

export const MyModule = new GraphQLModule({
  typeDefs,
  resolvers
});
```

**Comment by lewebsimple** [comment]
@AhmedElywa you might have to tweak things to get the prisma client instance in the resolvers context as graphql-modules uses DI / injector.
I will have time to test this later this week.
Thanks for the awesome tools, can't wait to try them out !

**Comment by AhmedElywa** [comment]
ok let me how you add prisma to context and how i add resolvers middleware to add select function like this one 

```ts
const middleware = async (resolve, root, args, context: Context, info) => {
  await context.prisma.connect();
  context.select = getPrismaSelect(info);
  const result = await resolve(root, args, context, info);
  await context.prisma.disconnect();
  return result;
};
```

**Comment by ardatan** [comment]
@AhmedElywa If you use DI, you can use `onRequest` and `onResponse` hooks. If not, you can use resolversComposition.

**Comment by AhmedElywa** [comment]
@ardatan @lewebsimple Finally i do it. 
It take very hard work because it's first time to deal with modules 

Tool
https://github.com/AhmedElywa/prisma-tools/tree/master/packages/graphql-modules

Example
https://github.com/AhmedElywa/prisma-tools/tree/master/examples/graphql-modules

@ardatan you need to add this to your docs it's get me 3 hours search about this solution 
https://medium.com/the-guild/manage-circular-imports-hell-with-graphql-modules-4b1611dee781

you can add my example inside your repo examples 

Thanks

**Comment by AhmedElywa** [comment]
@ardatan if you mentioned my tool inside your docs will be great thanks

**Comment by lewebsimple** [comment]
Awesome work @AhmedElywa !! I was just about to test out your tools when I noticed you just pushed the graphql-modules integration 👍

**Comment by kamilkisiela** [comment]
Closing in favor of #1466
